### PR TITLE
Add me.leucoso.Grock

### PR DESCRIPTION
**Pull request by dnlburgess**
Grock is an application to display geological maps of the UK and associated geological data.

### CODE REVIEW

File: me.leucoso.Grock.json [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf",
+        "--system-talk-name=org.freedesktop.GeoClue2"
+    ],
+    "build-options": {
+        "cflags": "-O2 -g",
+        "cxxflags": "-O2 -g",
+        "env": {
+            "V": "1"
+        }
+    },
```
Comment (bilelmoussaoui) [inline me.leucoso.Grock.json:24]:
```suggestion
```
already set by default

File: me.leucoso.Grock.json [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    "command": "grock",
+    "finish-args": [
+        "--share=network",
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf:ro",
+        "--talk-name=ca.desrt.dconf",
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf",
```
Comment (bilelmoussaoui) [inline me.leucoso.Grock.json:15]:
```suggestion
```
shouldn't be needed anymore

File: me.leucoso.Grock.json [hunk 3]
```diff
@@ -0,0 +1,86 @@
+{
+    "app-id": "me.leucoso.Grock",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.36",
+    "sdk": "org.gnome.Sdk",
+    "command": "grock",
+    "finish-args": [
+        "--share=network",
+        "--share=ipc",
+        "--socket=x11",
```
Comment (bilelmoussaoui) [inline me.leucoso.Grock.json:10]:
```suggestion
        "--socket=fallback-x11",
```

File: me.leucoso.Grock.json [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        "*.a"
+    ],
+    "modules": [
+    	{
+            "name" : "libchamplain",
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://codeberg.org/dburgess/libchamplain-wms.git"
```
Comment (bilelmoussaoui) [inline me.leucoso.Grock.json:43]: Use a tagged release here please or a specific commit

File: me.leucoso.Grock.json [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+            "buildsystem" : "meson",
+            "builddir" : true,
+            "config-opts" : [
+                "--libdir=/app/lib",
+                "-Denable-gtk-doc=false"
+            ],
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://gitlab.gnome.org/GNOME/geocode-glib.git"
```
Comment (bilelmoussaoui) [inline me.leucoso.Grock.json:58]: Use a tagged release here please or a specific commit. You can also use a tarball form download.gnome.org/sources/

File: me.leucoso.Grock.json [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+                }
+            ]
+        },
+        {
+            "name" : "proj",
+            "buildsystem" : "autotools",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/OSGeo/proj"
```
Comment (bilelmoussaoui) [inline me.leucoso.Grock.json:68]: Same as other modules

File: me.leucoso.Grock.json [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+        {
+            "name": "grock",
+            "builddir": true,
+            "buildsystem": "meson",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://codeberg.org/dburgess/Grock.git",
+                    "tag": "v1.0.0",
+		    "commit": "e23fe894d0beca5a7221edc6036985ab19d3b24c"
```
Comment (bilelmoussaoui) [inline me.leucoso.Grock.json:81]: Nitpick: indentation

File: me.leucoso.Grock.json [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+                    "url" : "https://codeberg.org/dburgess/libchamplain-wms.git"
+                }
+            ]
+        },
+        {
+            "name" : "geocode-glib",
+            "buildsystem" : "meson",
+            "builddir" : true,
+            "config-opts" : [
+                "--libdir=/app/lib",
```
Comment (bilelmoussaoui) [inline me.leucoso.Grock.json:52]:
```suggestion
```
pretty sure this is set by default?

File: me.leucoso.Grock.desktop [hunk 9]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
```
Comment (bilelmoussaoui) [inline me.leucoso.Grock.desktop:1]: Should be upstream if possible

File: me.leucoso.Grock.appdata.xml [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+  <categories>
+    <category>Geology</category>
+    <category>Geoscience</category>
+    <categroy>Maps</categroy>
+    <category>Geography</category>
+    <category>DataVisualization</category>
+    <category>Education</category>
+    <category>Science</category>
+    <category>GTK</category>
+  </categories>
```
Comment (bilelmoussaoui) [inline me.leucoso.Grock.appdata.xml:22]: The categories from the appdata & desktop files are merged together at a later stage of the build. You can just mention them on the desktop file.

File: me.leucoso.Grock.appdata.xml [hunk 11]
```diff
@@ -0,0 +1,65 @@
+  <?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+	<id>me.leucoso.Grock.desktop</id>
+	<metadata_license>CC0-1.0</metadata_license>
+	<project_license>GPL-3.0-or-later</project_license>
+  <name>Grock</name>
```
Comment (bilelmoussaoui) [inline me.leucoso.Grock.appdata.xml:6]: Nitpick: indentation

File: me.leucoso.Grock.appdata.xml [hunk 12]
```diff
@@ -0,0 +1,65 @@
+  <?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+	<id>me.leucoso.Grock.desktop</id>
```
Comment (bilelmoussaoui) [inline me.leucoso.Grock.appdata.xml:3]:
```suggestion
	<id>me.leucoso.Grock</id>
```

File: me.leucoso.Grock.appdata.xml [hunk 13]
```diff
@@ -0,0 +1,65 @@
+  <?xml version="1.0" encoding="UTF-8"?>
```
Comment (bilelmoussaoui) [inline me.leucoso.Grock.appdata.xml:1]:
```suggestion
<?xml version="1.0" encoding="UTF-8"?>
```

**Comment by bilelmoussaoui** [comment]
The appdata file should be upstreamed as well, ideally it should be renamed to use `.metainfo.xml` suffix and installed under `$datadir/metainfo` per the specs.

**Comment by bilelmoussaoui** [comment]
bot, build me.leucoso.Grock

**Comment by dnlburgess** [comment]
bot, build me.leucoso.Grock

**Comment by dnlburgess** [comment]
bot, build me.leucoso.Grock

File: me.leucoso.Grock.json [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+                    "type" : "git",
+                    "url" : "https://gitlab.gnome.org/GNOME/geocode-glib.git",
+                    "tag" : "3.26.2",
+                    "commit" : "34f420cbe1ff3f45fde9179ba07e0259b9ce505b"
+                }
+            ]
+        },
+        {
+            "name" : "proj",
+            "buildsystem" : "autotools",
```
Comment (nedrichards) [inline me.leucoso.Grock.json:56]: nit: this isn't required. autotools is the default.

**Comment by nedrichards** [comment]
You need the `--device=dri` permission:

> nick@nedr-yoga:~$ flatpak run me.leucoso.Grock
> libGL error: MESA-LOADER: failed to retrieve device information
> libGL error: Version 4 or later of flush extension not found
> libGL error: failed to load driver: i915
> libGL error: failed to open /dev/dri/card0: No such file or directory
> libGL error: failed to load driver: i965

**Comment by nedrichards** [comment]
This is a super cool app, looking forward to adding it.

**Comment by dnlburgess** [comment]
Thank you very much!

bot, build me.leucoso.Grock

**Comment by dnlburgess** [comment]
bot, build me.leucoso.Grock

**Comment by dnlburgess** [comment]
bot, build me.leucoso.Grock

**Pull request closed without merging**